rasa/shared/nlu/training_data/formats/rasa_yaml.py

```diff
@@ -378,6 +378,23 @@ def dump(
             target: Name of the target object to write the YAML to.
             training_data: TrainingData object.
         """
+        result = self.training_data_to_yaml(training_data)
+
+        if result:
+            rasa.shared.utils.io.write_yaml(result, target, True)
+
+    def training_data_to_yaml(
```

Comment:
maybe `training_data_as_dict`? 
`training_data_to_yaml` sounds confusing, I'd expect it to return a YAML string, not a dict.

Reply:
This was to be consistent with `YAMLStoryWriter`, which has a method named `stories_to_yaml` that does something similar. But I can change this one since it is true that `dict` is more descriptive.